sale_order_product_recommendation/wizards/sale_order_recommendation.py

```diff
@@ -100,9 +105,13 @@ def _generate_recommendations(self):
             self.line_ids += new_line
             existing_product_ids.append(line.product_id.id)
         # Add those recommendations too
+        products_allowed_ids = self.env['product.product'].browse(
+            [l["product_id"][0] for l in found_lines]).filtered(
+            lambda x: x.active and x.sale_ok).ids
         i = 0
         for line in found_lines:
-            if line["product_id"][0] in existing_product_ids:
+            if (line["product_id"][0] in existing_product_ids or
```

Comment:
If you do the above, this `if` should change accordingly.

Reply:
```suggestion
            if line["product_id"][0] not in products_allowed_ids:
```

You don't need now the existing_product_ids check because it's applied already in the domain above.